src/resmom/mom_server.c

```diff
@@ -1462,7 +1477,7 @@ state_to_server(void)
 
 	DBPRT(("updating state 0x%x to server\n", i))
 
-	if (vnlp != NULL)
+	if ((vnlp != NULL) && (mom_only == 0))
```

Comment:
Instead of comparing mom_only with 0, please use UPDATE_VNODES. 
Suggestion - may be change the variable name itself from "mom_only" to "update_what"